common/src/View/Actions.h

```diff
@@ -105,6 +96,47 @@ namespace TrenchBroom {
             Action& operator=(Action&& other) = default;
         };
 
+        /**
+         * ExecuteFn has type ActionExecutionContext& -> void
+         * EnabledFn has type ActionExecutionContext& -> bool
+         * CheckedFn has type ActionExecutionContext& -> bool
+         */
+        template <class ExecuteFn, class EnabledFn, class CheckedFn>
+        class LambdaAction : public Action {
+        private:
+            ExecuteFn m_execute;
+            EnabledFn m_enabled;
+            CheckedFn m_checked;
+            bool m_checkable;
+        public:
+            LambdaAction(const IO::Path& preferencePath, const QString& label, ActionContext::Type actionContext, const QKeySequence& defaultShortcut,
+                const ExecuteFn& execute, const EnabledFn& enabled, const CheckedFn& checked, const bool checkable, const IO::Path& iconPath)
```

Comment:
I wonder if it would not be better to take the lambdas as rvalue references and move them instead of copying them
around. Probably won't make much of a difference, but it looks like the copies are unnecessary.

Reply:
I tried but couldn't get it to compile.
Btw I have read that in this pattern for functions that take lambdas:
template<class T>
void something(T&& x){}

x is a universal reference not an rvalue reference. This stuff is a bit fuzzy to me though.

Reply:
I think for x to be a universal reference, T must be a template parameter of the function, not the class, because universal references are tied to function template argument deduction. So in this case, it would be an rvalue reference. If you are ok with it, I'd like to give it a try, maybe I can get it working.

Reply:
I have ended up not using rvalue references, instead we now take the parameters by value and std::move them into the LambdaAction constructor, which also takes the parameters by value and std::moves them into the fields. 

On the call site, the arguments are all temporaries anyway, so they are rvalues.